### ISSUE TITLE
Verify that the core utils work with musl-libc

### ISSUE THREAD

**Issue by Member1221** [dunex-project/dunex-core#64]
To fully replace gnu libs, we should avoid glibc via using musl-libc.
We should verify that all our applications work with musl; if not, then we should adjust accordingly.

**Comment by moon-chilled** [comment]
Or make a new libc.  I was [working on one](https://github.com/moon-chilled/dlibc) at one point, but I've been busy with school so haven't had as much time to work on it.

**Comment by Member1221** [comment]
Yeah no, we'll want to prefer something like musl, because we can verify that it's maintained more thoroughly. This project is not a full-time effort.

**Comment by moon-chilled** [comment]
Not suggesting necessarily to use my libc; just saying that making one is a possibility.

**Comment by chaomodus** [comment]
Personally I'm not opposed to attacking the problem of writing libC but that is, in itself,  a ridiculously large project with its own set of complexities, and if there's already a good full portable libc we can use it would probably be preferable.

**Comment by Marie-Joseph** [comment]
I've begun working on fixing compatibility with musl libc. For my own benefit, here are a list of programs which currently fail to compile under musl, a description of the error, and a first-glance-first-idea solution:

### sleep - sleep.d
* Error: "core.sys.posix.unistd import usleep not found"
* Solution: use `core.time.Duration` and `core.thread.osthread.sleep`
* Note: this needs to be ported to the commandr framework as well

### su - su.d
* Error: fails to build dependency `dunex-auth`
* Solution: fix `dunex-auth`

### dunex-auth - crypt.d
* Error: "core.sys.posix.unistd import crypt not found"
* Solution: use a crypto library for hashing? or at least copy a freely-licensed algorithm
* Note: I've opened an [issue on dunex-auth](https://github.com/dunex-project/dunex-auth/issues/4)

### sync - sync.d
* Error: "core.sys.posix.unistd import sync not found"
* Solution: access `sync` directly

### uname - uname.d
* Error: "no property `version_` for type `core.sys.posix.sys.utsname.utsname`"
* Solution: use property `version`
